src/EmailRecoveryManager.sol

```diff
@@ -380,16 +388,18 @@ contract EmailRecoveryManager is EmailAccountRecovery, Initializable, IEmailReco
 
         recoveryRequest.currentWeight += guardianStorage.weight;
 
-        uint256 threshold = guardianConfigs[account].threshold;
-        if (recoveryRequest.currentWeight >= threshold) {
+        if (recoveryRequest.currentWeight >= guardianConfig.threshold) {
+            bytes32 calldataHash = IEmailRecoverySubjectHandler(subjectHandler)
+                .parseRecoveryCalldataHash(templateIdx, subjectParams);
```

Comment:
Was this function separated from validateRecoverySubject to return calldataHash efficiently?

Reply:
Yeah it was a recommendation from the auditors to reduce unnecessary computation every time `processRecovery` was called